doc/help/elektra-data-structures.md

```diff
@@ -366,3 +366,80 @@ The algorithm uses the `syncbits` to decide if the key set needs to be
 synchronised.
 
 Continue reading [with the error handling](elektra-error-handling.md).
+
+## Order Preserving Minimal Perfect Hash Map
+
+### Vstack
+
+The `Vstack` is a virtual stack implementation with dynamical memory allocation.
+The allocation doubles the space if the `Vstack` is full and reduces the space by the half
+if the memory usage gets under a quarter.
+
+The data will be stored as void pointer and can be inserted with the push call.
+Retrieve is possible through the pop call.
+Besides the classical init and del functions, an is empty check is also implemented.
+
+The typedef `Vstack` contains the following fields:
+
+	typedef struct
+	{
+		size_t minSize;
+		size_t size;
+		void ** data;
+		void ** head;
+	} Vstack;
```

Comment:
I am not sure if this duplication is a good idea. Did you try to use tracability links to the header file and source files instead?